backend/src/schema/resolvers/searches/queryString.js

```diff
@@ -0,0 +1,47 @@
+export function queryString(str) {
+  const normalizedString = normalizeWhitespace(str)
+  const escapedString = escapeSpecialCharacters(normalizedString)
+  return `
+${matchWholeText(escapedString)}
+${matchEachWordExactly(escapedString)}
+${matchSomeWordsExactly(escapedString)}
+${matchBeginningOfWords(escapedString)}
+`
+}
+
+const matchWholeText = (str, boost = 8) => {
+  return `"${str}"^${boost}`
+}
+
+const matchEachWordExactly = (str, boost = 4) => {
+  if (!str.includes(' ')) return ''
+  const tmp = str
+    .split(' ')
+    .map((s, i) => (i === 0 ? `"${s}"` : `AND "${s}"`))
+    .join(' ')
+  return `(${tmp})^${boost}`
+}
+
+const matchSomeWordsExactly = (str, boost = 2) => {
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-03-16T09:24:14Z" title="Monday, March 16th 2020, 10:24:14 am +01:00">Mar 16, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,61 @@
+export function queryString(str) {
+  const normalizedString = normalizeWhitespace(str)
+  const escapedString = escapeSpecialCharacters(normalizedString)
+  return `
+${matchWholeText(escapedString)}
+${matchEachWordExactly(escapedString)}
+${matchSomeWordsExactly(escapedString)}
+${matchBeginningOfWords(escapedString)}
+`
+}
+
+const matchWholeText = (str, boost = 8) => {
+  return `"${str}"^${boost}`
+}
+
+const matchEachWordExactly = (str, boost = 4) => {
+  if (str.includes(' ')) {
+    const tmp = str
+      .split(' ')
+      .map((s, i) => (i === 0 ? `"${s}"` : `AND "${s}"`))
+      .join(' ')
+    return `(${tmp})^${boost}`
+  } else {
+    return ''
+  }
```


```
if (!str.includes(' ')) return ''
const tmp = str
  .split(' ')
  .map((s, i) => (i === 0 ? `"${s}"` : `AND "${s}"`))
  .join(' ')
return `(${tmp})^${boost}`
```

Reply:
> [<img alt="Mogge" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-03-17T10:53:49Z" title="Tuesday, March 17th 2020, 11:53:49 am +01:00">Mar 17, 2020</time>_

---


Thanks. This is much better